examples/rvv_reduce.c

```diff
@@ -30,9 +30,9 @@ void reduce(double *a, double *b, double *result_sum, int *result_count,
     vfloat64m1_t vec_a = __riscv_vle64_v_f64m1(a, vl);
     vfloat64m1_t vec_b = __riscv_vle64_v_f64m1(b, vl);
 
-    vbool64_t mask = __riscv_vmfne_vv_f64m1_b64(vec_a, vec_zero, vl);
+    vbool64_t mask = __riscv_vmfne_vf_f64m1_b64(vec_a, 42, vl);
 
-    vec_s = __riscv_vfmacc_vv_f64m1_m(mask, vec_s, vec_a, vec_b, vl);
+    vec_s = __riscv_vfmacc_vv_f64m1_tum(mask, vec_s, vec_a, vec_b, vl);
```

Comment:
Supposedly we shouldn't need the tail elements because we are doing reduction for the given `n` elements right?

Reply:
No, the reduction happens outside of the loop, so the last iteration could cause elements we want to reduce later to b overwritten. Come to think of it, this should be `_tumu`.

Reply:
yeah, `_tumu` is needed. (as more clarification on `_tu` - this intrinsic will update the same `vec_s` with both `vl==VLMAX` and `vl<VLMAX`, and we want to keep the top `VLMAX-vl` items when transitioning from an iteration of the former to the latter,)

Reply:
qemu and spike should imho default to writing all ones to inactive elements in the agnostic policy, or at least add the option to enable this. It's way too easy to write and successfully test broken code. (I may look into this, but I haven't worked with the codebase before)

Reply:
It seems QEMU does support this: `-cpu rv64,v=on,vext_spec=v1.0,vlen=256,rvv_ta_all_1s=on,rvv_ma_all_1s=on`

Neat!

Reply:
Looks like @eopXD actually [added this](https://lists.nongnu.org/archive/html/qemu-riscv/2022-06/msg00050.html), great!
I only saw the code implementing it for load/stores when looking through the code, but this make rvv_reduce fail with just the `_m` variant.

Reply:
My question is resolved. Thank you guys!